### PR TITLE
Added stats

### PR DESCRIPTION
**Pull request by devgabrielcoman**
## Jira:
[2186](https://jira.superawesome.tv/secure/RapidBoard.jspa?rapidView=175&view=detail&selectedIssue=KWS-2186&quickFilter=762)

## Description:
Implemented the Stats / Events definitions from `Protobufs` into the `iOS Data SDK`.

This will allow us, technically, to log events from the UI to the Keen API.

Also added tests!

Things to look at:
- `EventsEnvironmentProtocol` - that adds a key value, because it's needed by Keen
- `StatsRequest` 
- `StatsServiceProvider`

## Image
N/A

### CODE REVIEW

File: Example/PopJamSDK_Tests/Stats/Stats+ObjectProviderTests.swift [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                done()
+            }
+        }
+    }
+    
+    private struct MockEvent: EventModelProtocol {
+        // empty
+    }
+    
+    private struct MockEventsEnvironment: EnventsEnvironmentProtocol {
```
Comment (GuilhermeMota93) [inline Example/PopJamSDK_Tests/Stats/Stats+ObjectProviderTests.swift]: Isn't this a duplicate from `Stats+Mocks.swift`? Can we not use that one instead?

File: PopJamSDK/Classes/Stats/Stats+ObjectProvider.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        else {
+            completionHandler (StatsError())
+        }
+    }
+    
+    // MARK: - Private
+    
+    private func process(f: @escaping (Error?) -> Void) -> ((Result<String>) -> Void) {
+        return { result in
```
Comment (GuilhermeMota93) [inline PopJamSDK/Classes/Stats/Stats+ObjectProvider.swift]: Being picky here, but this doesn't match what we have in the other providers.

Fair enough if this `process` was being used more than once, but in this case I was expecting to see this logic in the `future.onResult` as we've been doing so far. Also, `f` isn't the best naming for the parameter.

Again, being picky, not going to hold the PR approval over this one. Should wait on the other guys opinion.

Comment (Tbruni85) [reply PopJamSDK/Classes/Stats/Stats+ObjectProvider.swift]: I agree with @GuilhermeMota93 about the naming, `f` is definitely something we should change.

File: PopJamSDK/Classes/Stats/Stats+ObjectProvider.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            }
+        }
+    }
+    
+    private struct StatsError: PrintableErrorProtocol {
+        var message: String = "Could not initialise stats engine"
+    }
+}
+
+
```
Comment (Tbruni85) [inline PopJamSDK/Classes/Stats/Stats+ObjectProvider.swift]: space

File: PopJamSDK/Classes/Stats/Stats+ObjectProvider.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                break
+            }
+        }
+    }
+    
+    private struct StatsError: PrintableErrorProtocol {
+        var message: String = "Could not initialise stats engine"
+    }
+}
+
```
Comment (me2k) [inline PopJamSDK/Classes/Stats/Stats+ObjectProvider.swift]: Additional spaces here, remove them otherwise all good.

**Pull request merged**